broker/src/test/java/io/camunda/zeebe/broker/transport/adminapi/AdminApiRequestHandlerTest.java

```diff
@@ -145,4 +82,208 @@ private ServerOutput createServerOutput(
+
+      // when
+      final var responseFuture = handleRequest(request, handler);
+      scheduler.workUntilDone();
+
+      // then
+      assertThat(responseFuture)
+          .succeedsWithin(Duration.ofMinutes(1))
+          .matches(Either::isLeft)
+          .matches(Either::isLeft)
+          .extracting(Either::getLeft)
+          .extracting(ErrorResponse::getErrorCode)
+          .isEqualTo(ErrorCode.UNSUPPORTED_MESSAGE);
+    }
+  }
+
+  @Nested
+  @ExtendWith(MockitoExtension.class)
+  final class PauseExportingRequest {
+    @RegisterExtension
+    final ControlledActorSchedulerExtension scheduler = new ControlledActorSchedulerExtension();
+
+    private final PartitionAdminAccess adminAccess;
+    private final AdminApiRequestHandler handler;
+
+    public PauseExportingRequest(
+        @Mock(answer = Answers.RETURNS_SELF) final PartitionAdminAccess adminAccess,
+        @Mock final PartitionManagerImpl partitionManager,
+        @Mock final AtomixServerTransport transport) {
+      this.adminAccess = adminAccess;
+      when(partitionManager.getPartitionGroup()).thenReturn(mock(ManagedPartitionGroup.class));
+      when(partitionManager.getPartitions()).thenReturn(List.of());
+      when(partitionManager.createAdminAccess(any())).thenReturn(adminAccess);
+      handler = new AdminApiRequestHandler(transport, partitionManager);
+    }
+
+    @BeforeEach
+    void startHandler() {
+      scheduler.submitActor(handler);
+      scheduler.workUntilDone();
+    }
+
+    @Test
+    void shouldPauseExportingForGivenPartition() {
+      when(adminAccess.pauseExporting()).thenReturn(CompletableActorFuture.completed(null));
+
+      final var request = new AdminRequest();
+      request.setPartitionId(1);
+      request.setType(AdminRequestType.PAUSE_EXPORTING);
+
+      // when
+      final var responseFuture = handleRequest(request, handler);
+      scheduler.workUntilDone();
+
+      // then
+      assertThat(responseFuture).succeedsWithin(Duration.ofMinutes(1)).matches(Either::isRight);
+      verify(adminAccess).forPartition(request.getPartitionId());
+      verify(adminAccess).pauseExporting();
+    }
+
+    @Test
+    void shouldRespondWithFailureIfPausingFails() {
+      // given
+      when(adminAccess.pauseExporting())
+          .thenReturn(
+              CompletableActorFuture.completedExceptionally(
+                  new RuntimeException("Exporting fails")));
+
+      final var request = new AdminRequest();
+      request.setPartitionId(1);
+      request.setType(AdminRequestType.PAUSE_EXPORTING);
+
+      // when
+      final var responseFuture = handleRequest(request, handler);
+      scheduler.workUntilDone();
+
+      // then
+      assertThat(responseFuture)
+          .succeedsWithin(Duration.ofMinutes(1))
+          .matches(Either::isLeft)
+          .extracting(Either::getLeft)
+          .extracting(ErrorResponse::getErrorCode)
+          .isEqualTo(ErrorCode.INTERNAL_ERROR);
+    }
```

Comment:
:wrench: Extract this to a re-usable method. This is used multiple times to verify different error codes.